src/OmniSharp.Host/MSBuild/Discovery/Extensions.cs

```diff
@@ -18,7 +19,9 @@ public static void RegisterDefaultInstance(this IMSBuildLocator msbuildLocator,
                 else
                 {
                     instanceToRegister = instance;
-                    break;
+
+                    if (instance.HasDotNetSdksResolvers())
```

Comment:
So the idea here is that if we find an MSBuild that did come with the .NET SDK, we should stop searching?

Reply:
Correct. Now that I look a bit more closely, I guess the biggest difference is that if there aren't any VS instances that contain the MSBuildSdkResolver, we'll pick the last instance instead of the first instance. That should probably be fixed before merging to avoid any subtle breaking changes.

Reply:
Hey 👋 

So, I've rewrote it in a different manner. Which keeps the current behavior (preferring the first) but favors DotNet Core when comparing instances :)

I'll write more tests tomorrow and push it (If I have the time 😢)

Reply:
Can you please review it again? :)